### ISSUE TITLE
Add hook after org.eclipse.jetty.server.Server creation in ServerCommand#run

### ISSUE THREAD

**Issue by aantoniadis** [dropwizard/dropwizard#746]
I've already posted here https://groups.google.com/forum/#!topic/dropwizard-dev/G6N5rzUEKnI the reasons that I think such capability would be useful. Since Dropwizard exposes the Server through environment.getApplicationContext().getServer() I think that it would be reasonable to be able to access the Server before it is started in order to enable features like Distributed Session Clustering.

What do you think?

**Comment by aantoniadis** [comment]
any news on this issue?

**Comment by jplock** [comment]
Would you mind submitting a pull request for this functionality?

**Comment by aantoniadis** [comment]
@jplock I am currently working on the hooks. In the meantime I solved the reported problem by using subclassing the default server factory like the following:

```java
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
public class MyServerFactory extends DefaultServerFactory {
	
	private Server server;
	
	@Override
	protected Server buildServer(LifecycleEnvironment lifecycle,
			ThreadPool threadPool) {
		if (this.server == null) {
			this.server = super.buildServer(lifecycle, threadPool);
		}
		return this.server;
	}
	public Server getServer() {
		return this.server;
	}
}

Obviously this solution is not clean and leaks too much. I will come back with a PR on the hooks

**Comment by aantoniadis** [comment]
After some thinking on how should the hooks be implemented, I realized that they should be similar to lifecycle listeners. Actually, I found a response to my answer at StackOverflow (http://stackoverflow.com/questions/26054748/dropwizard-session-clustering).

Although lifecycle listeners seem to solve this particular problem, I believe that there should be either a more explicit way to install hooks or a more flexible way to manipulate the objects created (a bit more dependency-injection friendly). I believe that the latter may be better since it provides more flexibility in case somebody wants to do some customizations (in my case the biggest problem was that ServerCommand was created by Application and I couldn't change it or subclass it). 

What are your thoughts?
